src/UbiMesh.cpp

```diff
@@ -0,0 +1,320 @@
+/*
+Copyright (c) 2013-2019 Ubidots.
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+Developed and maintained by Jose Garcia for Ubidots Inc
+@jotathebest at github: https://github.com/jotathebest
+*/
+
+#include "UbiMesh.h"
+#include "UbiProtocolHandler.h"
+
+/**************************************************************************
+ * Static Functions
+ ***************************************************************************/
+#if PLATFORM_ID != PLATFORM_XENON && PLATFORM_ID != PLATFORM_XENON_SOM && \
+    PLATFORM_ID != PLATFORM_PHOTON_DEV &&                                 \
+    PLATFORM_ID != PLATFORM_PHOTON_PRODUCTION &&                          \
+    PLATFORM_ID != PLATFORM_ELECTRON_PRODUCTION
+bool UbiMesh::isThrottled() {
+  // Simple throttling routine for TCP and HTTP
+  bool throttled = true;
+  if (iotProtocolMesh == UBI_HTTP || iotProtocolMesh == UBI_TCP) {
+    if (millis() - time_now > 20000) {
+      throttled = false;
+      time_now = millis();
+    }
+  }
+
+  if (iotProtocolMesh == UBI_UDP || iotProtocolMesh == UBI_PARTICLE) {
+    throttled = false;
+  }
+
+  return throttled;
+}
+
+void UbiMesh::ubiPublishHandler(const char* event, const char* data) {
+  bool throttled = isThrottled();
+
+  if (throttled) {
+    if (_debugMesh) {
+      Serial.println(
+          "[WARNING] your sample time reached the throttling of 20 s, please "
+          "set a higher sample time or use UDP or Particle Webhooks to send "
+          "data");
+    }
+  }
+
+  if (!throttled) {
```

Comment:
change this for an `else` only if the previous conditional is not merged with the inner if.